backend/cmm_peephole_engine.mli

```diff
@@ -0,0 +1,60 @@
+type _ pattern_kind =
+  | Expr : Cmm.expression pattern_kind
+  | Int : int pattern_kind
+  | Natint : Nativeint.t pattern_kind
+
+type 'a pattern_var
+
+val create_var : 'a pattern_kind -> string -> 'a pattern_var
+
+module Default_variables : sig
+  val c : Cmm.expression pattern_var
+
+  val c1 : Cmm.expression pattern_var
+
+  val c2 : Cmm.expression pattern_var
+
+  val n : int pattern_var
+
+  val n1 : int pattern_var
+
+  val n2 : int pattern_var
+end
+
+module Env : sig
+  type t
+end
+
+type binop =
+  | Add
+  | Sub
+  | Lsl
+  | Lsr
+  | Asr
+  | Or
+  | And
+  | Comparison
+  | Bitwise_op
```

Comment:
I think that for future rules, we'll likely need more precise binops for comparisons and bitwise operations, what's your plan for then ? add new binops ?

Reply:
Yes, those types are expanded as needed. `Bitwise_op` already subsumes `Or` and `And`; the constructors do not have to match distinct expressions.